src/core/layout/qgslayoutitemmapgrid.cpp

```diff
@@ -1894,6 +1895,22 @@ bool QgsLayoutItemMapGrid::shouldShowForDisplayMode( QgsLayoutItemMapGrid::Annot
          || ( mode == QgsLayoutItemMapGrid::LongitudeOnly && coordinate == QgsLayoutItemMapGrid::Longitude );
 }
 
+
+QgsLayoutItemMapGrid::DisplayMode gridAnnotationDisplayModeFromDD( QString ddValue, QgsLayoutItemMapGrid::DisplayMode defValue )
+{
+  if ( ddValue == QStringLiteral( "x_only" ) )
```

Comment:
Can you make these comparisons case insensitive? Also requires some unit tests for this.

Reply:
done